cmd/kubeadm/app/apis/kubeadm/v1alpha1/types.go

```diff
@@ -74,8 +74,13 @@ type Etcd struct {
 type NodeConfiguration struct {
 	metav1.TypeMeta `json:",inline"`
 
-	Discovery  Discovery `json:"discovery"`
-	CACertPath string    `json:"caCertPath"`
+	CACertPath        string   `json:"caCertPath"`
+	DiscoveryURL      string   `json:"discoveryUrl"`
+	DiscoveryFile     string   `json:"discoveryFile"`
+	DiscoveryToken    string   `json:"discoveryToken"`
+	TLSBootstrapToken string   `json:"tlsBootstrapToken"`
+	Token             string   `json:"token"`
+	Masters           []string `json:"masters"`
```

Comment:
As long as we are going to have to remove the command line flag, we really should change this to `DiscoveryTokenAPIServers` to be correct and move it up to be right after `DiscoveryToken`